### DISCUSSION TITLE
Testing strategies: unit, integration, and end-to-end

### DISCUSSION THREAD

**Discussion by jrschumacher** [connectrpc/.github#14] [General]
I would love to get a discussion going around testing strategies.

---

Topics on connectrpc.com[^search-results]:
- Go: _TBD_
- Web: https://connectrpc.com/docs/web/testing/
- Node.js: https://connectrpc.com/docs/node/testing/
- Swift: https://connectrpc.com/docs/swift/testing/
- Kotlin: _TBD_
- curl & other: _TBD_ (cURL is self-explanatory)
- End-to-end test frameworks:
	- Cypress: https://connectrpc.com/docs/web/supported-browsers-and-frameworks/#cypress

[^search-results]: https://connectrpc.com/search?q=test

**Comment by timostamm** [comment]
[Playwright](https://connectrpc.com/docs/web/testing/#end-to-end-testing) is a nice alternative to Cypress for e2e testing.
